stdlib/public/core/SmallString.swift

```diff
@@ -231,13 +231,13 @@ extension _SmallString: RandomAccessCollection, MutableCollection {
 extension _SmallString {
   @inlinable @inline(__always)
   @safe
-  internal func withUTF8<Result>(
-    _ f: (UnsafeBufferPointer<UInt8>) throws -> Result
-  ) rethrows -> Result {
+  internal func withUTF8<Success, Failure: Error>(
+    _ f: (UnsafeBufferPointer<UInt8>) throws(Failure) -> Success
+  ) throws(Failure) -> Success {
```

Comment:
`@inlinable` exports an ABI symbol; this change removes that symbol and introduces a new one, without a proper availability declaration. I expect we'll need to mark the new declaration `@backDeployed` or `@_alwaysEmitIntoClient`, and we'll need a SwiftStdlibLegacyABI entry point for the old ABI.